src/main/java/com/google/sps/utility/FreeTimeUtility.java

```diff
@@ -0,0 +1,174 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.utility;
+
+import com.google.sps.data.CalendarDataResponse;
+import java.util.*;
+import java.util.concurrent.TimeUnit;
+
+/** Class containing the user's free hours. */
+public final class FreeTimeUtility {
```

Comment:
Food for thought: I wonder if we could simplify this a bit, particularly share some logic among the different cases.
One way I thought of:
1) You could have a function `int calcIntersection(Interval event, Interval window)` that would return the intersection in hours between event and window. Then, for each event `e`, you would call `calcIntersection(e, personalWindow)` and `calcIntersection(e, workWindow)`.
2) In order to take into account the fact that events can have intersections among themselves, you could augment `calcIntersection` and make it return an intersection interval `[a, b]` and memorize those. After you are done processing all the events, you would look into those intervals and calculate the remaining free time.

Those are just ideas, you don't need to strictly follow them, but I think it's worth spending some time trying to use something along those lines to make this class a bit simpler. Please let me know if you need any clarification/help.

Reply:
I used a similar idea but rather calculated directly the remaining free interval based on the intervals for each event. The core method is similar to the one used in the fifth walkthrough.

Reply:
The overall algorithm is now much easier to understand - thank you.

Reply:
I agree it looks much cleaner. Thanks for working on this!

Reply:
RESOLVED